handling-forms/controlled-forms.md

```diff
@@ -226,50 +226,61 @@ Sofia's app currently has the following components:
 ```js
 import { useState } from 'react';
 import StudentList from './components/StudentList';
+import ClassInfo from './components/ClassInfo';
 
 function App() {
-    const [studentData, setStudentData] = useState([
-        {
-            id: 1,
-            nameData: 'Ada',
-            emailData: 'ada@dev.org',
-            isPresentData: false
-        },
-        {
-            id: 2,
-            nameData: 'Soo-ah',
-            emailData: 'sooah@dev.org',
-            isPresentData: false
-        },
-        {
-            id: 3,
-            nameData: 'Chrissy',
-            emailData: 'chrissy@dev.org',
-            isPresentData: true
-        }
-    ]);
-
-    const updateStudentData = updatedStudent => {
-        const students = studentData.map(student => {
-            if (student.id === updatedStudent.id) {
-                return updatedStudent;
-            } else {
-                return student;
-            }
-        });
-
-        setStudentData(students);
-    };
+  const [studentData, setStudentData] = useState([
+    {
+      id: 1,
+      nameData: 'Ada',
+      emailData: 'ada@dev.org',
+      isPresentData: false,
+    },
+    {
+      id: 2,
+      nameData: 'Soo-ah',
+      emailData: 'sooah@dev.org',
+      isPresentData: false,
+    },
+    {
+      id: 3,
+      nameData: 'Chrissy',
+      emailData: 'chrissy@dev.org',
+      isPresentData: true,
+    },
+  ]);
+
+  const toggleStudentPresence = (studentId) => {
+    // calculate the updated student data by finding the student that matches
+    // the passed id, making a copy with object spreading, then overwriting
+    // the presence value with its inverse
+    const students = studentData.map(student => {
+      if (student.id === studentId) {
+        // this was the toggled student, so make a new record with the updated
+        // presence value
+        return { ...student, isPresentData: !student.isPresentData };
+      } else {
+        // this was not the student who was toggled, so we can use the existing
+        // data in the new student array
+        return student;
+      }
+    });
 
-    return (
-        <main>
-            <h1>Attendance</h1>
-            <StudentList
-                students={studentData}
-                onUpdateStudent={updateStudentData}
-                ></StudentList>
-        </main>
-    );
+    // uses value-passing style to update the student data, but could be
+    // refactored to use function-passing style
+    setStudentData(students);
+  };
+
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <ClassInfo memberCount={studentData.length}></ClassInfo>
+      <StudentList
+        students={studentData}
+        onStudentPresenceToggle={toggleStudentPresence}
+      ></StudentList>
+    </main>
+  );
```

Comment:
Pulled this code from: https://github.com/AdaGold/student-app/blob/05-lifting-state-up-npm-id/src/App.jsx

Reply:
Though the repo code is using 2 spaces, Learn was using 4, so when I copied the new code into other lessons I ran a quick search/replace over the code lines to replace all occurrences of 2 spaces with 4 spaces